interface/modules/custom_modules/oe-module-comlink-telehealth/moduleConfig.php

```diff
@@ -5,13 +5,102 @@
  *  package   OpenEMR
  *  link      http://www.open-emr.org
  *  author    Sherwin Gaddis <sherwingaddis@gmail.com>
+ *  author    Jerry Padgett <sjpadgett@gmail.com>
  *  copyright Copyright (c )2021. Sherwin Gaddis <sherwingaddis@gmail.com>
+ *  copyright Copyright (c) 2023 Jerry Padgett <sjpadgett@gmail.com>
  *  license   https://github.com/openemr/openemr/blob/master/LICENSE GNU General Public License 3
  *
  */
 
 require_once dirname(__FILE__, 4) . '/globals.php';
 
+use OpenEMR\Common\Csrf\CsrfUtils;
+
 $module_config = 1;
 
-require_once('welcome.php');
+if (!empty($_GET['setup']) ?? null) {
+    /**
+     * Example expected data structure for received data.
+     * $args = ['ctsiVBData' => [
+     * 'registrationUri' => 'https://sginsts.comlinktelehealth.io:28749/CTSIVB/',
+     * 'videoApiUri' => 'https://sginsts.comlinktelehealth.io:22528',
+     * 'ctsiOrgUid' => 'OPEN5C0D4F',
+     * 'ctsiOrgPwd' => 'DD71C92B-834D-4C',
+     * 'ctsiOrgId' => 'OPENEM280D0D']];
+     *
+     * */
+    if (!CsrfUtils::verifyCsrfToken($_GET["csrf_token_form"])) {
+        CsrfUtils::csrfNotVerified();
+    }
+
+    $content = trim(file_get_contents("php://input"));
+    $credentials = json_decode($content, true);
+
+    $items['comlink_telehealth_registration_uri'] = $credentials['registrationUri'];
+    $items['comlink_telehealth_video_uri'] = $credentials['videoApiUri'];
+    $items['comlink_telehealth_user_id'] = $credentials['ctsiOrgUid'];
+    $items['comlink_telehealth_user_password'] = $credentials['ctsiOrgPwd'];
+    $items['comlink_telehealth_cms_id'] = $credentials['ctsiOrgId'];
+    // Save to globals table.
+    foreach ($items as $key => $credential) {
+        sqlQuery(
+            "INSERT INTO `globals` (`gl_name`,`gl_value`) VALUES (?, ?) ON DUPLICATE KEY UPDATE `gl_name` = ?, `gl_value` = ?",
+            array($key, $credential, $key, $credential)
+        );
+    }
+    // Recall GLOBALS to verify the save.
+    require_once dirname(__FILE__, 4) . "/globals.php";
```

Comment:
this is already require_once above, so above line would be ignored. Also shouldn't bring in globals.php twice in same script or weird things can happen. Prob shouldn't need to verify this anyways since you are doing doing this sql insert above (and if wanted to confirm, could just grab it from globals table with a sql query).

Reply:
I meant require.

Reply:
I found this interesting. I always suspected require_once does not work with globals in many scopes which this proved me correct. Though I meant to require, it still works. My bet is that since globals has some dynamic components on init it may fool interpreter. It also explains some of the scope issues we see from time to time. Or could be the windows compiled version.
Or could be my aging brain:)